core/src/main/java/com/yahoo/ycsb/generator/UniformGenerator.java

```diff
@@ -48,7 +48,7 @@ public UniformGenerator(Collection<String> values)
 	@Override
   public String nextValue()
 	{
-		_laststring = _values.get(_gen.nextValue());
+		_laststring = _values.get(_gen.nextValue().intValue());
```

Comment:
Is there a reason we are doing .intValue()? Sorry I wasn't clear earlier with my comment. Should we be using longs everywhere?